server/models/User.ts

```diff
@@ -308,6 +308,30 @@ class User extends ParanoidModel<
 
   // instance methods
 
+  /**
+   * Whether the user belongs to a given team.
+   *
+   * @param team The team to check
+   * @returns Whether the user belongs to the team
+   */
+  public isMemberOf(team?: Team | string | null) {
+    if (!team) {
+      return false;
+    }
+    const teamId = typeof team === "string" ? team : team.id;
+    return this.teamId === teamId;
+  }
+
+  /**
+   * Whether the user is a given team's admin.
+   *
+   * @param team The team to check
+   * @returns Whether the user is an admin of the team
+   */
+  public isTeamAdmin(team?: Team | string | null) {
```

Comment:
Aware of another `isTeamAdmin` under `policies/utils`. Open to thoughts on which one should we stick to...

Reply:
Would prefer to keep these methods under `policies`